hw/top_earlgrey/cdc/cdc_waivers.tcl

```diff
@@ -50,6 +50,131 @@ set_rule_status -rule {W_GLITCH} -status {Waived}          \
     (GlitchOutput=~"*u_sync_1*")} \
   -comment {Waive PADs input goes into synchronizer}
 
+# Driving from PAD RESET to SPI TPM
+set_rule_status -rule {W_ASYNC_RST_FLOPS} -status {Waived}                           \
+  -expression {(DrivingSignal=~"IO*") && (ReceivingFlop=~"*u_spi_device.u_spi_tpm.*")} \
+  -comment {PAD RESET driving to SPI TPM}
+
+set_rule_status -rule {W_ASYNC_RST_FLOPS} -status {Waived}                           \
+  -expression {(DrivingSignal=~"SPI_DEV_CS_L*") && (ReceivingFlop=~"*u_spi_device.u_spi_tpm.*")} \
+  -comment {PAD RESET driving to SPI TPM}
+
+set_rule_status -rule {W_ASYNC_RST_FLOPS} -status {Waived}                           \
+  -expression {(DrivingSignal=~"top_earlgrey.u_pinmux_aon*") && (ReceivingFlop=~"*u_spi_device.u_spi_tpm.*")} \
+  -comment {PAD RESET driving to SPI TPM}
+
+set_rule_status -rule {W_ASYNC_RST_FLOPS} -status {Waived}                           \
+  -expression {(DrivingSignal=~"top_earlgrey.u_rstmgr_aon*") && (ReceivingFlop=~"*u_spi_device.u_spi_tpm.*")} \
+  -comment {PAD RESET driving to SPI TPM}
+
+# Driving from PAD PIN MUX RESET to DMI JTAG
+set_rule_status -rule {W_ASYNC_RST_FLOPS} -status {Waived}                           \
+  -expression {(DrivingSignal=~"top_earlgrey.u_pinmux_aon*") && (ReceivingFlop=~"*u_dmi_jtag.*")} \
+  -comment {PAD RESET driving to dmi jtag}
+
+# dual port memory to SPI
+set_rule_status -rule {W_MASYNC} -status {Waived}                           \
+  -expression {(Driver =~ "top_earlgrey.u_spi_device.u_memory_2p.u_mem.gen_generic.u_impl_generic.b_rdata_o*") && (ReceivingFlop=~"top_earlgrey.u_spi_device.u_fwmode.*")} \
+  -comment {Dual port memory read port to SPI}
+
+set_rule_status -rule {W_MASYNC} -status {Waived}                           \
+  -expression {(Driver =~ "top_earlgrey.u_spi_device.u_control_abort.q*") && (ReceivingFlop=~"top_earlgrey.u_spi_device.u_fwmode.*")} \
+  -comment {Dual port memory read port to SPI}
+
+# retention regs (#13811)
+set_rule_status -rule {W_MASYNC} -status {Waived}                           \
+  -expression {(Driver =~ "SPI_DEV_*") && (ReceivingFlop=~"top_earlgrey.u_pinmux_aon.dio_oe_retreg_q*")} \
+  -comment {retention regs}
+
+# retention regs (#13811)
+set_rule_status -rule {W_MASYNC} -status {Waived}                           \
+  -expression {(Driver =~ "SPI_HOST_*") && (ReceivingFlop=~"top_earlgrey.u_pinmux_aon.dio_oe_retreg_q*")} \
+  -comment {retention regs}
+
+# PAD to spi_host
+set_rule_status -rule {W_MASYNC} -status {Waived}                           \
```

Comment:
are these because of retention registers also..? is it the retention registers (in the pinmux clock domain) being fed back to spi_host? is that the error?

Reply:
Yes, they are retention registers which receive and mux multiple paths from different clock domains in the pinmux block. When I asked for clarifying, I got the answer that these were semi-static.

Reply:
sounds good! just like above, could you add a header comment? That will be super helpful.